### PR TITLE
"Save settings" token acquired fix.  azure calls errors partially pro…

### PR DESCRIPTION
**Pull request by bergano65**
"Save settings" token acquired fix.  azure calls errors partially proceeded

### CODE REVIEW

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 1]
```diff
@@ -12,7 +12,6 @@
 
 namespace CogsMinimizer.Controllers
 {
-    [Authorize]
```
Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: I don't understand, why did you remove the [Authorize] requirement.
Every Action in this controller needs to be done by an authenticated user.

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 2]
```diff
@@ -12,7 +12,6 @@
...
     public class SubscriptionController : SubMinimizerController
```
Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: Please add validation to each one of the public actions that the parameters passed to it are valid (not null or empty) otherwise, throw an exception

File: CogsMinimizer/Controllers/SubscriptionController.cs (hunk 1 above)
Comment (bergano65) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: none of controller classes have it.

Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: SubscriptionController does.
Home/Index checks for Authenticated user at the beginning of the method.

The other pages (About, Error) do not require authentication.

The settings page should add it as well.

File: CogsMinimizer/Controllers/HomeController.cs [hunk 3]
```diff
@@ -84,6 +84,8 @@ public ActionResult Index()
 
         public ActionResult Error(string Exception)
         {
+            Diagnostics.EnsureStringNotNullOrWhiteSpace(() => Exception);
```
Comment (shklar) [inline CogsMinimizer/Controllers/HomeController.cs]: I have done some telemetry modifications to the Error Action.
Please sync first.
